src/sentry/migrations/0519_add_flat_file_index_table.py

```diff
@@ -0,0 +1,87 @@
+    # schema of a table.
+    # Here are some things that make sense to mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that they can
+    #   be monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
+    is_dangerous = False
+
+    dependencies = [
+        ("sentry", "0518_cleanup_bundles_indexes"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="ArtifactBundleFlatFileIndex",
+            fields=[
+                (
+                    "id",
+                    sentry.db.models.fields.bounded.BoundedBigAutoField(
+                        primary_key=True, serialize=False
+                    ),
+                ),
+                (
+                    "project_id",
+                    sentry.db.models.fields.bounded.BoundedBigIntegerField(db_index=True),
+                ),
+                ("release_name", models.CharField(max_length=250)),
+                ("dist_name", models.CharField(default="", max_length=64)),
+                ("date_added", models.DateTimeField(default=django.utils.timezone.now)),
+                (
+                    "flat_file_index",
+                    sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        null=True, on_delete=django.db.models.deletion.CASCADE, to="sentry.file"
+                    ),
+                ),
+            ],
+            options={
+                "db_table": "sentry_artifactbundleflatfileindex",
+                "index_together": {("project_id", "release_name", "dist_name")},
+            },
+        ),
+        migrations.CreateModel(
+            name="FlatFileIndexState",
+            fields=[
+                (
+                    "id",
+                    sentry.db.models.fields.bounded.BoundedBigAutoField(
+                        primary_key=True, serialize=False
+                    ),
+                ),
+                ("indexing_state", models.IntegerField()),
+                ("date_added", models.DateTimeField(default=django.utils.timezone.now)),
+                (
+                    "artifact_bundle_id",
+                    sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        on_delete=django.db.models.deletion.CASCADE, to="sentry.artifactbundle"
+                    ),
+                ),
+                (
+                    "flat_file_index_id",
+                    sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        on_delete=django.db.models.deletion.CASCADE,
+                        to="sentry.artifactbundleflatfileindex",
+                    ),
+                ),
```

Comment:
You might end up with a column name of `flat_file_index_id_id` here as Django generally expects foreignkey fields to have the relation name without the `_id` suffix on it.

Reply:
Oh yeah, my bad, overlooked this, both for index and artifact bundle. Thanks @markstory